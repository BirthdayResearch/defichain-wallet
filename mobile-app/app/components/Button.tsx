import { TouchableOpacity, TouchableOpacityProps } from 'react-native'
import { useThemeContext } from '@shared-contexts/ThemeProvider'
import { tailwind } from '@tailwind'
import { Text } from './Text'
import { ThemedActivityIndicator } from '@components/themed'

export type ButtonColorType = 'primary' | 'secondary'
export type ButtonFillType = 'fill' | 'outline' | 'flat'

interface ButtonProps extends React.PropsWithChildren<TouchableOpacityProps> {
  color?: ButtonColorType
  fill?: ButtonFillType
  label?: string
  margin?: string
  title?: string
  isSubmitting?: boolean
  submittingLabel?: string
}

export function Button (props: ButtonProps): JSX.Element {
  const {
    label,
    submittingLabel,
    color = 'primary',
    fill = 'fill',
    margin = 'm-4 mt-8',
    isSubmitting = false
  } = props
  const { isLight } = useThemeContext()
  const themedColor = isLight ? `${color}` : `dark${color}`

  const disabledStyle = isLight ? 'bg-gray-200 border-0' : 'bg-gray-600 text-gray-500 border-0'
  const disabledText = isLight ? 'text-gray-400' : 'text-gray-500'

  const buttonColor = isLight ? `bg-${themedColor}-50` : `bg-${themedColor}-700`
  const buttonStyle = `${fill === 'fill' ? buttonColor : 'bg-transparent'}`
  const buttonText = isLight ? `text-${themedColor}-500` : `${fill === 'fill' ? 'text-white' : 'text-primary-700'}`

  const textStyle = `${props.disabled === true ? disabledText : buttonText} ${isSubmitting ? 'ml-2' : ''}`
  const text = isSubmitting ? submittingLabel ?? label : label

  return (
    <TouchableOpacity
      {...props}
      style={tailwind(`${margin} p-3 rounded flex-row justify-center ${buttonStyle} ${props.disabled === true ? disabledStyle : ''}`)}
    >
      {
        text !== undefined && (
          <>
            {isSubmitting && <ThemedActivityIndicator />}
            <Text style={(tailwind(`${textStyle} font-bold text-center`))}>
              {text}
            </Text>
          </>
        )
      }

      {
        props.children
      }
    </TouchableOpacity>
  )
}
