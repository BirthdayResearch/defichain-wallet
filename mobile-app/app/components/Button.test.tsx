import { fireEvent, render } from '@testing-library/react-native'
import { Text } from 'react-native'
import { Button, ButtonColorType, ButtonFillType } from './Button'

jest.mock('@shared-contexts/ThemeProvider')

const buttonFill: ButtonFillType[] = ['fill', 'outline', 'flat']
const buttonColor: ButtonColorType[] = ['primary', 'secondary']

describe('button', () => {
  buttonFill.forEach(fill => {
    buttonColor.forEach(color => {
      it(`should match styling of button type ${fill}-${color}`, () => {
        const onPress = jest.fn()
        const enabled = render(
          <Button
            color={color}
            disabled={false}
            fill={fill}
            label='Submit'
            onPress={onPress}
            title='Test'
          />).toJSON()
        expect(enabled).toMatchSnapshot()

        const disabled = render(
          <Button
            color={color}
            disabled
            fill={fill}
            label='Submit'
            onPress={onPress}
            title='Test'
          />).toJSON()
        expect(disabled).toMatchSnapshot()

        const submitting = render(
          <Button
            color={color}
            disabled
            isSubmitting
            submittingLabel='Submitting'
            fill={fill}
            label='Submit'
            onPress={onPress}
            title='Test'
          />).toJSON()
        expect(submitting).toMatchSnapshot()
      })
    })
  })

  it('should be clickable', async () => {
    const onPress = jest.fn()
    const component = (
      <Button
        onPress={onPress}
        testID='primary_button'
        title='Submit'
      >
        <Text>
          Hello World
        </Text>
      </Button>
    )
    const rendered = render(component)
    const receiveButton = await rendered.findByTestId('primary_button')
    fireEvent.press(receiveButton)
    expect(rendered.toJSON()).toMatchSnapshot()
    expect(onPress).toBeCalled()
  })

  it('should not be clickable when disabled', async () => {
    const onPress = jest.fn()
    const component = (
      <Button
        disabled
        onPress={onPress}
        testID='primary_button'
        title='Submit'
      >
        <Text>
          Hello World
        </Text>
      </Button>
    )
    const rendered = render(component)
    const receiveButton = await rendered.findByTestId('primary_button')
    fireEvent.press(receiveButton)
    expect(rendered.toJSON()).toMatchSnapshot()
    expect(onPress).toBeCalledTimes(0)
  })
})
