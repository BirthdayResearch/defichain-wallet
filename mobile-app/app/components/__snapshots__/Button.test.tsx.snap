// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should be clickable 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 229, 247, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
  testID="primary_button"
>
  <Text>
    Hello World
  </Text>
</View>
`;

exports[`button should match styling of button type fill-primary 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 229, 247, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(255, 0, 175, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type fill-primary 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type fill-primary 3`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ff00af"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "marginLeft": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Submitting
  </Text>
</View>
`;

exports[`button should match styling of button type fill-secondary 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(232, 222, 255, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(91, 15, 255, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type fill-secondary 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type fill-secondary 3`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ff00af"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "marginLeft": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Submitting
  </Text>
</View>
`;

exports[`button should match styling of button type flat-primary 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(255, 0, 175, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type flat-primary 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type flat-primary 3`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ff00af"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "marginLeft": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Submitting
  </Text>
</View>
`;

exports[`button should match styling of button type flat-secondary 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(91, 15, 255, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type flat-secondary 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type flat-secondary 3`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ff00af"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "marginLeft": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Submitting
  </Text>
</View>
`;

exports[`button should match styling of button type outline-primary 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(255, 0, 175, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type outline-primary 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type outline-primary 3`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ff00af"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "marginLeft": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Submitting
  </Text>
</View>
`;

exports[`button should match styling of button type outline-secondary 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(91, 15, 255, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type outline-secondary 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "textAlign": "center",
        },
      ]
    }
  >
    Submit
  </Text>
</View>
`;

exports[`button should match styling of button type outline-secondary 3`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ff00af"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Object {
          "color": "rgba(163, 163, 163, 1)",
          "fontFamily": "BoldFont",
          "fontWeight": "700",
          "marginLeft": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Submitting
  </Text>
</View>
`;

exports[`button should not be clickable when disabled 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(229, 229, 229, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
  testID="primary_button"
>
  <Text>
    Hello World
  </Text>
</View>
`;
